data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamAcknowledgementManager.java

```diff
@@ -95,19 +97,23 @@ private void monitorAcknowledgment(final ExecutorService executorService, final
                     } else {
                         LOG.debug("Checkpoint not complete for resume token {}", checkpointStatus.getResumeToken());
                         final Duration ackWaitDuration = Duration.between(Instant.ofEpochMilli(checkpointStatus.getCreateTimestamp()), Instant.now());
+                        // negative ack
                         if (checkpointStatus.isNegativeAcknowledgement()) {
-                            // Give up partition and should interrupt parent thread to stop processing stream
-                            if (lastCheckpointStatus != null && lastCheckpointStatus.isPositiveAcknowledgement()) {
-                                checkpoint(lastCheckpointStatus.getResumeToken(), lastCheckpointStatus.getRecordCount());
-                            }
-                            LOG.warn("Acknowledgement not received for the checkpoint {} past wait time. Giving up partition.", checkpointStatus.getResumeToken());
-                            partitionCheckpoint.giveUpPartition();
-                            this.giveupPartitionCount.increment();
+                            LOG.warn("Negative Acknowledgement received for the checkpoint {}. Giving up partition.", checkpointStatus.getResumeToken());
+                            giveUpPartition(lastCheckpointStatus);
                             break;
+                        } else {
+                            // no ack received within timeout period (15 secs + 60 secs)
+                            if (!ackWaitDuration.minus(partitionAcknowledgmentTimeout)
+                                    .minusMillis(checkPointIntervalInMs).isNegative()) {
+                                LOG.warn("Acknowledgement not received for the checkpoint {} past wait time. Giving up partition.", checkpointStatus.getResumeToken());
+                                giveUpPartition(lastCheckpointStatus);
```

Comment:
Do you think it's better to give up in this case?

Reply:
When no ack is received after timeout, then not sure if we will ever receive it. So we need to restart processing the partition from checkpoint. Updated the no ack timeout to 15 mins, so take care of extreme sink backed up case.